tests/msc3083_test.go

```diff
@@ -93,6 +93,15 @@ func checkRestrictedRoom(t *testing.T, alice *client.CSAPI, bob *client.CSAPI, s
 	// Leaving the room works and the user is unable to re-join.
 	bob.LeaveRoom(t, room)
 	bob.LeaveRoom(t, space)
+
+	alice.SyncUntilTimelineHas(t, space, func(ev gjson.Result) bool {
```

Comment:
I wonder if having `LeaveRoom` is a mistake, and we should always be doing the synced form..

Reply:
Maybe, though the subtlety here is that we need to wait for it to be processed on the *remote* server, rather than the local one.